index.js

```diff
@@ -56,24 +140,30 @@ async function fastifyBasicAuth (fastify, opts) {
   }
 }
 
-function getAuthenticateHeader (authenticate) {
-  if (!authenticate) return false
+function getAuthenticateHeader (authenticate, useUtf8) {
```

Comment:
I don't think it is a good idea to create infinite amount of function.

Reply:
Its not creating an infinite amount of functions. The function generated by getAuthenticateHeader is assigned once in line 76
https://github.com/fastify/fastify-basic-auth/blob/889cc95787755ae4077b2c4c10a5aa30c4ddb102/index.js#L76
and then reused by the plugin instance.